maps/ministation/ministation_objects.dm

```diff
@@ -14,6 +19,35 @@
 		if(!.)
 			. = SSmaterials.weighted_minerals_rich
 
+/turf/exterior/wall/random/ministation/Initialize(var/ml, var/materialtype, var/rmaterialtype)
+	..(ml, materialtype, rmaterialtype)
+	if(ml)
+		starting = TRUE
+	. = INITIALIZE_HINT_LATELOAD
+
+/turf/exterior/wall/random/ministation/LateInitialize()
+	if(starting) //only need to call generate_asteroid here if this is a first node, the others are called in the generate_asteroid function
+		if(prob(99) && !guaranteed) //99% chance that the asteroid turns into space
+			ChangeTurf(/turf/space)
+		else //otherwise, try to convert the turfs in the cardinal direction into asteroid as well
+			generate_asteroid(generatepercent)
+	. = ..()
+
+//tries to convert the space turfs around the asteroid into asteroids as well and tries to keep already existing asteroids around it from deleting in mapload.
+/turf/exterior/wall/random/ministation/proc/generate_asteroid(var/probability)
+	var/turf/exterior/wall/random/ministation/newasteroid
+	for(var/ndir in global.cardinal)
+		var/turf/ad = get_step(src, ndir)
+		if(prob(probability))
+			if(istype(ad, /turf/space)) //if it's space turn it into asteroid
+				newasteroid = ad.ChangeTurf(/turf/exterior/wall/random/ministation)
+				newasteroid.generate_asteroid(max(10,probability-10)) //reduce the probability of conversion with 10 percent for each turf away from the starting one.
+			if(istype(ad, /turf/exterior/wall/random/ministation)) //if it's asteroid prevent it from being deleted in it's own LateInitialize proc.
```

Comment:
This feels a bit strange - aren't you going to run into ordering issues where you're modifying the vars on an asteroid that has already fired its LateInitialize()? With the space change, aren't you going to get cases of double-calling generate_asteroid()? Apologies, it's a bit hard to follow the expected chain of events here.

Reply:
If the wall is already initialized the modification of vars have no effect. generate_asteroid() on an already existing wall will only be called by it's LateInitialize method so no double calling. However it's true that it could be solved more clearly and elegantly so brb.